Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,453 @@
+    , sosDeviceModel = DeviceProduct
+    , DvcOs = "SonicOS"
+    , DvcOsVersion = EventProductVersion
+    , DvcIdType = "Other"
+    , Dvc = sosSerialNumber
+    , DvcDescription = DeviceProduct
+    , ASimMatchingHostname = "-"
+    , ASimMatchingIpAddr = "-"
+    , NetworkIcmpType = tostring(NetworkIcmpType)
+    , NetworkIcmpCode = toint(NetworkIcmpCode)
+    , NetworkBytes = tolong(coalesce(toint(ReceivedBytes), 0) + coalesce(toint(SentBytes), 0))
+    , FieldDeviceCustomNumber1Label = column_ifexists("FieldDeviceCustomNumber1Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , FieldDeviceCustomNumber2Label = column_ifexists("FieldDeviceCustomNumber2Label", "") // FieldDeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber1Label = column_ifexists("DeviceCustomNumber1Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber2Label = column_ifexists("DeviceCustomNumber2Label", "") // DeviceCustomNumberXLabel (cnXLabel=)
+    , DeviceCustomNumber3Label = column_ifexists("DeviceCustomNumber3Label", "") // Connection Duration in milliseconds (for Connnection Closed msgs)
+    , fw_action = column_ifexists("fw_action", "")
+    , sosFirewallAction = column_ifexists("fw_action", "")
+    , susr = column_ifexists("susr", "") // Logged-in username associated with the log event.
+    , af_polid = column_ifexists("af_polid", "")
+    , af_action = column_ifexists("af_action", "")
+    , af_policy = column_ifexists("af_policy", "")
+    , af_service = column_ifexists("af_service", "")
+    , af_type = column_ifexists("af_type", "")
+    , af_object = column_ifexists("af_object", "")
+    , contentObject = column_ifexists("contentObject", "") // App Firewall/App Rule Content.
+    , srcV6 = column_ifexists("srcV6", "") // Source IPv6 IP
+    , dstV6 = column_ifexists("dstV6", "") // Destination IPv6 IP
+    , appName = column_ifexists("appName", "") // Non-signature Application name that matches the Application ID "app" or "f".
+    , app = column_ifexists("app", "") // Numeric Application ID. Not the same as "ApplicationProtocol".
+    , sosAppCategoryName = extract(@'^["]([a-zA-Z-]+)', 1, column_ifexists("appcat", "")) // Application Category from App Control
+    , sosAppIDName = extract(@' (.*) --', 1, column_ifexists("appcat", "")) // Application Name from App Control
+    , sosAppSignatureName = trim('"', extract(@'.*-- (.*\S)', 1, column_ifexists("appcat", ""))) // Application Signature Name from App Control
+    , catid = column_ifexists("catid", "") // Application Category ID
+    , sid = column_ifexists("sid", "") // Application Signature ID
+    , appid = column_ifexists("appid", "") // Application ID from App Control
+    , sosIPSFullString = column_ifexists("ipscat", "")
+    , ipscat = extract(@'^"?([a-zA-Z-\/]+)', 1, column_ifexists("ipscat", "")) // IPS Category/Signature
+    , sosIPSSignatureName = extract(@'[ ](.*)\S', 1, column_ifexists("ipscat", "")) // IPS Signature name

```

Comment:
ThreatName?

Reply:
Mapping them to schema fields, thank you.